cmd/autothrottle/main.go

```diff
@@ -175,25 +182,46 @@ func main() {
 		log.Fatal(err)
 	}
 
-	throttleMeta := &ReplicationThrottleConfigs{
+	ThrottleManager := &ThrottleManager{
 		zk:                     zk,
 		km:                     km,
+		kafkaNativeMode:        Config.KafkaNativeMode,
+		kafkaAPIRequestTimeout: Config.KafkaAPIRequestTimeout,
 		events:                 events,
 		previouslySetThrottles: make(replicationCapacityByBroker),
 		limits:                 lim,
 		failureThreshold:       Config.FailureThreshold,
 	}
 
+	// Init a KafkaAdmin Client if needed.
+	if Config.KafkaNativeMode {
+		if err := ThrottleManager.InitKafkaAdmin(Config.BootstrapServers); err != nil {
+			log.Fatal(err)
+		}
+		log.Printf("Connected to Kafka: %s\n", Config.BootstrapServers)
+	}
+
 	// Run.
 	var interval int64
 	var ticker = time.NewTicker(time.Duration(Config.Interval) * time.Second)
 
 	// TODO(jamie): refactor this loop.
-	for {
+	for ; ; <-ticker.C {
 		interval++
 
 		// Get topics undergoing reassignment.
-		reassignments = zk.GetReassignments() // XXX This needs to return an error.
+		if !Config.KafkaNativeMode {
+			reassignments = zk.GetReassignments()
+		} else {
+			// KIP-455 compatible reassignments lookup.
+			reassignments, err = zk.ListReassignments()
+		}
+
+		if err != nil {
```

Comment:
should this error only be checked on `KafkaNativeMode`? i.e, put inside the above else branch.

Reply:
Pulled this into the conditional